### PR TITLE
chore(BACK-8009): update slices support

### PR DESCRIPTION
**Pull request by jnicoulaud-ledger**
Update following https://github.com/LedgerHQ/clear-signing-erc7730-registry/pull/83
- start and end are optional
- end is exclusive

### CODE REVIEW

**Review by fsamier — APPROVED** [review]

File: src/erc7730/model/paths/__init__.py [hunk 1]
```diff
@@ -67,24 +67,30 @@ class ArraySlice(Model):
...
     )
 
     @model_validator(mode="after")
     def _validate(self) -> Self:
-        if self.start > self.end:
-            raise ValueError("Array slice start index must be lower than end index.")
+        if (start := self.start) is None or (end := self.end) is None:
+            return self
+        if 0 <= end <= start or end <= start < 0:
+            raise ValueError("Array slice start index must be strictly lower than end index.")
```
Comment (ckorchane-ledger) [inline src/erc7730/model/paths/__init__.py:89]: I think we should test if start and end have the same sign, otherwise this passes: start = -2, end = 1

Comment (fsamier) [reply src/erc7730/model/paths/__init__.py:89]: In Python, which seems to be the inspiration, it's valid to have different signs, but indeed that does not seem to be the intent of the specification.

Comment (ckorchane-ledger) [reply src/erc7730/model/paths/__init__.py:89]: I just wanted to make sure this returns an error to the user: `array = [23, 21, 10, 14], start = -2, end = 1` where obiously it is a typo in the input.

Comment (jnicoulaud-ledger) [reply src/erc7730/model/paths/__init__.py:89]: You can't tell without the array, so `[-2:1]` is invalid for `[23, 21, 10, 14]`, but it's valid for `[23, 21, 14]` => therefore it is valid. There are valid use cases for this, eg `4:-20` for bytearray like `selector || variable size stuff || address`

**Pull request merged**